presto-bytecode/src/main/java/com/facebook/presto/bytecode/DumpBytecodeVisitor.java

```diff
@@ -0,0 +1,615 @@
+                invokeInstruction.getMethodDescription());
+        return null;
+    }
+
+    @Override
+    public Void visitInvokeDynamic(BytecodeNode parent, InvokeDynamicInstruction invokeDynamicInstruction)
+    {
+        printLine("invokeDynamic %s%s %s",
+                invokeDynamicInstruction.getName(),
+                invokeDynamicInstruction.getMethodDescription(),
+                invokeDynamicInstruction.getBootstrapArguments());
+        return null;
+    }
+
+    //
+    // Control Flow
+    //
+
+    @Override
+    public Void visitTryCatch(BytecodeNode parent, TryCatch tryCatch)
+    {
+        if (tryCatch.getComment() != null) {
+            printLine();
+            printLine("// %s", tryCatch.getComment());
+        }
+
+        printLine("try {");
+        indentLevel++;
+        tryCatch.getTryNode().accept(tryCatch, this);
+        indentLevel--;
+        printLine("}");
+
+        printLine("catch (%s) {", tryCatch.getExceptionName());
+        indentLevel++;
+        tryCatch.getCatchNode().accept(tryCatch, this);
+        indentLevel--;
+        printLine("}");
+
+        return null;
+    }
+
+    @Override
+    public Void visitIf(BytecodeNode parent, IfStatement ifStatement)
+    {
+        if (ifStatement.getComment() != null) {
+            printLine();
+            printLine("// %s", ifStatement.getComment());
+        }
+        printLine("if {");
+        indentLevel++;
+        visitNestedNode("condition", ifStatement.condition(), ifStatement);
+        if (!ifStatement.ifTrue().isEmpty()) {
+            visitNestedNode("ifTrue", ifStatement.ifTrue(), ifStatement);
+        }
+        if (!ifStatement.ifFalse().isEmpty()) {
+            visitNestedNode("ifFalse", ifStatement.ifFalse(), ifStatement);
+        }
+        indentLevel--;
+        printLine("}");
+        return null;
+    }
+
+    @Override
+    public Void visitFor(BytecodeNode parent, ForLoop forLoop)
+    {
+        if (forLoop.getComment() != null) {
+            printLine();
+            printLine("// %s", forLoop.getComment());
+        }
+        printLine("for {");
+        indentLevel++;
+        visitNestedNode("initialize", forLoop.initialize(), forLoop);
+        visitNestedNode("condition", forLoop.condition(), forLoop);
+        visitNestedNode("update", forLoop.update(), forLoop);
+        visitNestedNode("body", forLoop.body(), forLoop);
+        indentLevel--;
+        printLine("}");
+        return null;
+    }
+
+    @Override
+    public Void visitWhile(BytecodeNode parent, WhileLoop whileLoop)
+    {
+        if (whileLoop.getComment() != null) {
+            printLine();
+            printLine("// %s", whileLoop.getComment());
+        }
+        printLine("while {");
+        indentLevel++;
+        visitNestedNode("condition", whileLoop.condition(), whileLoop);
+        visitNestedNode("body", whileLoop.body(), whileLoop);
+        indentLevel--;
+        printLine("}");
+        return null;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 365:375, 381:391.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.